docs/src/main/sphinx/admin/properties-optimizer.rst

```diff
@@ -117,6 +117,15 @@ the maximum number of joins that can be reordered at once.
 Reduces number of rows produced by joins when optimizer detects that duplicated
 join output rows can be skipped.
 
+``optimizer.use-exact-partitioning``
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+* **Type:** :ref:`prop-type-boolean`
+* **Default value:** ``false``
+
+Re-partitions data if the partitioning of the input isn't exactly what is
```

Comment:
```suggestion
When enabled this forces data repartitioning unless the partitioning of upstream stage matches exactly what downstream stage expects
```

Input and output are not really accurate.

BTW, do we need to mention the session property `use_exact_partitioning` too?